scroll_derive/src/lib.rs

```diff
@@ -166,32 +174,38 @@ fn impl_try_into_ctx(
     });
     let gn = quote! { #gl #( #gn ),* #gg };
     let gwref = if !gp.is_empty() {
-        let gi = gp.iter().map(|param: &syn::GenericParam| match param {
+        let gi: Vec<_> = gp.iter().filter_map(|param: &syn::GenericParam| match param {
             syn::GenericParam::Type(ref t) => {
                 let ident = &t.ident;
-                quote! {
+                Some(quote! {
                     &'a #ident : ::scroll::ctx::TryIntoCtx<::scroll::Endian>,
                     ::scroll::Error: ::std::convert::From<<&'a #ident as ::scroll::ctx::TryIntoCtx<::scroll::Endian>>::Error>,
                     <&'a #ident as ::scroll::ctx::TryIntoCtx<::scroll::Endian>>::Error: ::std::convert::From<scroll::Error>
-                }
+                })
             },
-            p => quote! { #p }
-        });
-        quote! { where #( #gi ),* }
+            syn::GenericParam::Lifetime(_) => None,
+            p => Some(quote! { #p }),
+        }).collect();
+        if !gi.is_empty() {
+            quote! { where #( #gi ),* }
+        } else {
+            quote! {}
+        }
     } else {
         quote! {}
     };
     let gw = if !gp.is_empty() {
-        let gi = gp.iter().map(|param: &syn::GenericParam| match param {
+        let gi = gp.iter().filter_map(|param: &syn::GenericParam| match param {
             syn::GenericParam::Type(ref t) => {
                 let ident = &t.ident;
-                quote! {
+                Some(quote! {
                     #ident : ::scroll::ctx::TryIntoCtx<::scroll::Endian>,
                     ::scroll::Error: ::std::convert::From<<#ident as ::scroll::ctx::TryIntoCtx<::scroll::Endian>>::Error>,
                     <#ident as ::scroll::ctx::TryIntoCtx<::scroll::Endian>>::Error: ::std::convert::From<scroll::Error>
-                }
+                })
             },
-            p => quote! { #p }
+            syn::GenericParam::Lifetime(_) => None,
```

Comment:
not clear to me why this branch is `None` here

Reply:
actually 'the heart of the fix' is here.
the prevoius code doesn't process lifetime properly, which will generates tokens like
```
impl<'b> ... for ... 
where
    ... : ... ,
    'b
    ^^ invalid syntax
```

Reply:
i see, thank you for clarifying!